storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -2194,10 +2195,23 @@ bool copy_back(int argc, char **argv) {
   if (srv_undo_dir && *srv_undo_dir && !directory_exists(srv_undo_dir, true)) {
     return (false);
   }
+
+  sprintf(metadata_path, "%s/%s", xtrabackup_target_dir,
+          XTRABACKUP_METADATA_FILENAME);
   if (innobase_data_home_dir && *innobase_data_home_dir &&
       !directory_exists(innobase_data_home_dir, true)) {
     return (false);
   }
+  if (!xtrabackup_read_metadata(metadata_path)) {
+    msg("xtrabackup: Error: failed to read metadata from '%s'\n",
+        metadata_path);
+    exit(EXIT_FAILURE);
+  }
```

Comment:
I think exiting is not right. Maybe throw a warning and proceed with the old behavior.

Reply:
It is added because in 8.0 we can't start the server which is not fully prepared.

Reply:
Are you sure? Can you try a full backup and try restoring without prepare?

Reply:
I tried but I can't reproduce the case which caused copy back to fail if it is just prepare without apply-log. So what to do you suggest just add warning?

Reply:
Yeah, at least a warning for a few releases before making it throw an error.